src/packets.c

```diff
@@ -1020,6 +1063,156 @@ TbBool process_players_dungeon_control_packet_action(long plyr_idx)
     case PckA_ToggleComputer:
         toggle_computer_player(plyr_idx);
         break;
+    case PckA_CheatPlaceTerrain:
+    {
+        x = ((unsigned short)pckt->pos_x);
+        y = ((unsigned short)pckt->pos_y);
+        stl_x = coord_subtile(x);
+        stl_y = coord_subtile(y);
+        slb_x = subtile_slab_fast(stl_x);
+        slb_y = subtile_slab_fast(stl_y);
+        if (slab_kind_is_animated(pckt->actn_par1))
+        {
+            place_animating_slab_type_on_map(pckt->actn_par1, 0, stl_x, stl_y, pckt->actn_par2);
+        }
+        else
+        {
+            place_slab_type_on_map(pckt->actn_par1, stl_x, stl_y, pckt->actn_par2, 0);
+        }
+        do_slab_efficiency_alteration(slb_x, slb_y);
+        break;
+    }
+    case PckA_CheatMakeCreature:
+    {
+        x = ((unsigned short)pckt->pos_x);
+        y = ((unsigned short)pckt->pos_y);
+        pos.x.val = x;
+        pos.y.val = y;
+        PlayerNumber id = pckt->actn_par2;
+        unsigned char exp = pckt->actn_par2 >> 8;
+        thing = create_creature(&pos, pckt->actn_par1, id);
+        if (!thing_is_invalid(thing))
+        {
+            set_creature_level(thing, exp);
+        }
+        break;
+    }
+    case PckA_CheatMakeDigger:
+    {
+        x = ((unsigned short)pckt->pos_x);
+        y = ((unsigned short)pckt->pos_y);
+        thing = create_owned_special_digger(x, y, pckt->actn_par1);
+        if (!thing_is_invalid(thing))
+        {
+            set_creature_level(thing, pckt->actn_par2);
+        }
+        break;
+    }
+    case PckA_CheatStealSlab:
+    {
+        x = ((unsigned short)pckt->pos_x);
+        y = ((unsigned short)pckt->pos_y);
+        stl_x = coord_subtile(x);
+        stl_y = coord_subtile(y);
+        slb_x = subtile_slab_fast(stl_x);
+        slb_y = subtile_slab_fast(stl_y);
+        PlayerNumber id = pckt->actn_par2;
+        TbBool effect = pckt->actn_par2 >> 8;
+        if (effect)
+        {
+            if (pckt->actn_par1 == SlbT_CLAIMED)
+            {
+                pos.x.val = subtile_coord_center(slab_subtile_center(subtile_slab(stl_x)));
+                pos.y.val = subtile_coord_center(slab_subtile_center(subtile_slab(stl_y))); 
+                pos.z.val = subtile_coord_center(1);
+                if (is_my_player(player))
+                {
+                    play_non_3d_sample(76);
+                }
+                create_effect(&pos, imp_spangle_effects[id], id);
+            }
+            else
+            {
+                if (is_my_player(player))
+                {
+                    play_non_3d_sample(41);
```

Comment:
41 is a magic number here, I know it's always done this way atm but best would be to add an enum somewhere or at a minimum add a comment saying what sound sound 41 is
applies for all occurrences of play_non_3d_sample ofcource